ui/ducks/send/send.js

```diff
@@ -3013,35 +3012,36 @@ export function signTransaction(history) {
         };
 
         if (bestQuote?.approvalNeeded) {
-          const { id } = await addTransactionAndWaitForPublish(
-            { ...bestQuote.approvalNeeded, amount: '0x0' },
-            {
-              requireApproval: false,
-              // TODO: create new type for swap+send approvals; works as stopgap bc swaps doesn't use this type for STXs in `submitSmartTransactionHook` (via `TransactionController`)
-              type: TransactionType.swapApproval,
-              swaps: {
-                hasApproveTx: true,
-                meta: {
-                  type: TransactionType.swapApproval,
-                  sourceTokenSymbol,
+          const { id } = await dispatch(
+            addTransactionAndRouteToConfirmationPage(
+              { ...bestQuote.approvalNeeded, amount: '0x0' },
+              {
+                requireApproval: true,
+                // TODO: create new type for swap+send approvals; works as stopgap bc swaps doesn't use this type for STXs in `submitSmartTransactionHook` (via `TransactionController`)
+                type: TransactionType.swapApproval,
+                swaps: {
+                  hasApproveTx: true,
+                  meta: {
+                    type: TransactionType.swapApproval,
+                    sourceTokenSymbol,
+                  },
                 },
               },
-            },
+            ),
           );
           meta.approvalTxId = id;
         }
 
-        const { id: swapAndSendTxId } = await addTransactionAndWaitForPublish(
-          txParams,
-          {
-            requireApproval: false,
+        const { id: swapAndSendTxId } = await dispatch(
+          addTransactionAndRouteToConfirmationPage(txParams, {
+            requireApproval: true,
```

Comment:
Since we're now showing the confirmation, do we also want to run Blockaid validation as normal?

So should we update `SECURITY_PROVIDER_EXCLUDED_TRANSACTION_TYPES` in `shared/constants/security-provider.ts`?